### ISSUE TITLE
Is children aggregation working in version 1.4.2?

### ISSUE THREAD

**Issue by yangou** [elasticsearch/elasticsearch#9494]
I just upgraded my elasticsearch cluster to 1.4.2. I want to use the children aggregation for some statistics  use. Here is an example of my mappings:
for `users` type, I have a field called "gender"
```
{
   ...
    "gender":  "male",
   ...
}
```

the `users` type has a child type `events`, which has a field called `event_type`:
```
{
  "event_type": "purchase"
}
```

And here is my command for the aggregation:
```
curl -XGET 'http://localhost:9200/my_index/users/_search?search_type=count' -d '{
"aggs":{
  "genders": {
    "terms":{
      "field": "gender"
    },
    "aggs": {
      "events_aggs": {
        "children": {
           "type" : "events" 
        },
        "aggs": {
          "on_event_type": {
            "terms": {
              "field": "event_type",
              "size": 0
            }
          }
        }
      }
    }
  }
 } 
}'
```

And this always returns me empty bucket on the `on_event_type` level:
```
{"took":1812,"timed_out":false,"_shards":{"total":5,"successful":5,"failed":0},"hits":{"total":17814,"max_score":0.0,"hits":[]},"aggregations":{"genders":{"doc_count_error_upper_bound":0,"sum_other_doc_count":0,"buckets":[{"key":"unknown","doc_count":809,"events_aggs":{"doc_count":65459,"on_event_type":{"doc_count_error_upper_bound":0,"sum_other_doc_count":0,"buckets":[]}}},{"key":"female","doc_count":716,"events_aggs":{"doc_count":50830,"on_event_type":{"doc_count_error_upper_bound":0,"sum_other_doc_count":0,"buckets":[]}}},{"key":"male","doc_count":288,"events_aggs":{"doc_count":15762,"on_event_type":{"doc_count_error_upper_bound":0,"sum_other_doc_count":0,"buckets":[]}}}]}}}
```

Is my query correct? I was just doing in the same format as documented.

**Comment by yangou** [comment]
There are two version of documents talking about children aggregations. Both points to the `current`:
 version:
http://www.elasticsearch.org/guide/en/elasticsearch/guide/current/children-agg.html
and
http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-aggregations-bucket-children-aggregation.html

Maybe the second one is wrong in terms of the aggregation syntax.
The line  `"field": "owner.display_name"` should really be `"field": "answer.owner.display_name"`

I tried to change my aggregation syntax to the one mentioned in the first link, and it works.

**Comment by rjernst** [comment]
@yangou Yes, with 1.4, doing subaggs on the children aggregation requires specifying the type for each field.  However, this was just fixed yesterday (#9487) and will be in 1.5.

**Comment by yangou** [comment]
@rjernst Actually I think the current behavior is the meaningful. having the children type specified in the field does have some use cases.

For example, if I have a `user` type as parent, `event` type as children, and `user_activity` as another children. What if I want to aggregate on the children_type `event` and nest another aggregation on `user_activity`? Having the children_type specified would help on this.
